packages/test/service-load-test/src/loadTestDataStore.ts

```diff
@@ -18,113 +23,206 @@ export interface IRunConfig {
 export interface ILoadTest {
     run(config: IRunConfig): Promise<void>;
 }
-
 const wait = async (timeMs: number) => new Promise((resolve) => setTimeout(resolve, timeMs));
 
-class LoadTestDataStore extends DataObject implements ILoadTest {
-    public static DataStoreName = "StressTestDataStore";
-    private opCount = 0;
-    private sentCount = 0;
-    private state: string = "not started";
-    protected async hasInitialized() {
-        this.root.on("op", () => {
-            this.opCount++;
-        });
+class LoadTestDataStoreModel {
+    public static initializingFirstTime(root: ISharedDirectory, runtime: IFluidDataStoreRuntime) {
+        root.set("taskmanager", TaskManager.create(runtime).handle);
     }
 
-    public async pause(timeMs: number) {
-        const startTimeMs = Date.now();
-        this.state = "paused";
-        await wait(timeMs);
-        this.state = "running";
-        return Date.now() - startTimeMs;
+    private static async waitForCatchup(runtime: IFluidDataStoreRuntime) {
+        if(runtime.deltaManager.active) {
+            return;
+        }
+        const lastKnownSeq = runtime.deltaManager.lastKnownSeqNumber;
+        while(runtime.deltaManager.lastSequenceNumber < lastKnownSeq) {
```

Comment:
Should `lastSequenceNumber` be pulled off `deltaManager` again each turn of this loop?

Reply:
do you mean `lastKnownSeq`? fine to grab and reuse